packages/cactus-cmd-api-server/src/test/typescript/unit/get-open-api-spec-v1-oauth2-scopes.test.ts

```diff
@@ -0,0 +1,209 @@
+      unprotectedEndpointExemptions: [],
+      expressJwtOptions,
+      socketIoJwtOptions,
+      socketIoPath: Constants.SocketIoConnectionPathV1,
+    };
+
+    const pluginsPath = path.join(
+      __dirname,
+      "../../../../../../", // walk back up to the project root
+      ".tmp/test/test-cmd-api-server/get-open-api-spec-v1-endpoint_test/", // the dir path from the root
+      uuidv4(), // then a random directory to ensure proper isolation
+    );
+    const pluginManagerOptionsJson = JSON.stringify({ pluginsPath });
+
+    const pluginRegistry = new PluginRegistry({ logLevel });
+
+    const configService = new ConfigService();
+
+    const apiSrvOpts = await configService.newExampleConfig();
+    apiSrvOpts.logLevel = logLevel;
+    apiSrvOpts.pluginManagerOptionsJson = pluginManagerOptionsJson;
+    apiSrvOpts.authorizationProtocol = AuthorizationProtocol.JSON_WEB_TOKEN;
+    apiSrvOpts.authorizationConfigJson = authorizationConfig;
+    apiSrvOpts.configFile = "";
+    apiSrvOpts.apiCorsDomainCsv = "*";
+    apiSrvOpts.apiPort = 0;
+    apiSrvOpts.cockpitPort = 0;
+    apiSrvOpts.grpcPort = 0;
+    apiSrvOpts.crpcPort = 0;
+    apiSrvOpts.apiTlsEnabled = false;
+    apiSrvOpts.grpcMtlsEnabled = false;
+    apiSrvOpts.plugins = [];
+
+    const config = await configService.newExampleConfigConvict(apiSrvOpts);
+
+    apiServer = new ApiServer({
+      config: config.getProperties(),
+      pluginRegistry,
+    });
+
+    apiServer.initPluginRegistry({ pluginRegistry });
+    const startResponsePromise = apiServer.start();
+    await expect(startResponsePromise).toResolve();
+    const startResponse = await startResponsePromise;
+    expect(startResponse).toBeTruthy();
+
+    const { addressInfoApi } = await startResponsePromise;
+    const protocol = apiSrvOpts.apiTlsEnabled ? "https" : "http";
+    const { address, port } = addressInfoApi;
+    const apiHost = `${protocol}://${address}:${port}`;
+
+    const jwtPayload = { name: "Peter", location: "Albertirsa" };
+    const validJwt = await new SignJWT(jwtPayload)
+      .setProtectedHeader({ alg: "RS256" })
+      .setIssuer(expressJwtOptions.issuer)
+      .setAudience(expressJwtOptions.audience)
+      .sign(jwtKeyPair.privateKey);
+    expect(validJwt).toBeTruthy();
+
+    const validBearerToken = `Bearer ${validJwt}`;
+    expect(validBearerToken).toBeTruthy();
+
+    apiClient = new ApiServerApiClient(
+      new ApiServerApiClientConfiguration({
+        basePath: apiHost,
+        baseOptions: { headers: { Authorization: validBearerToken } },
+        logLevel,
+      }),
+    );
+  });
+
+  it("HTTP - returns the OpenAPI spec .json document of the API server itself", async () => {
+    const res1Promise = apiClient.getOpenApiSpecV1();
+    await expect(res1Promise).resolves.toHaveProperty("data.openapi");
+    const res1 = await res1Promise;
+    expect(res1.status).toEqual(200);
+    expect(res1.data).toBeTruthy();
```

Comment:
@aldousalvarez In this test case it is supposed to fail because you are executing with a JWT that does not have the scope declared (so the JWT has a valid signature but it does not declare the correct role based access control and therefore should not work)

Reply:
@aldousalvarez Thank you for the updates here as well!